src/beta/resources/referral.js

```diff
@@ -0,0 +1,137 @@
+      });
+    }
+
+    async save() {
+      if (this.phone_number !== undefined) {
+        // handle discrepancy between phone (create Referral) and phone_number (all User functions, retrieve Referral)
+        this.phone = this.phone_number;
+      }
+      return super.save();
+    }
+
+    /**
+     * Update the referral's email address.
+     * @param {string} referralUserId - The referral user's ID.
+     * @param {string} email - The new email address.
+     * @returns {Promise<boolean>}
+     */
+    static async updateEmail(referralUserId, email) {
+      try {
+        const newParams = { user: { email } };
+        await api.put(`${this._url}/${referralUserId}`, newParams);
+        return true;
+      } catch (e) {
+        return Promise.reject(e);
+      }
+    }
+
+    /**
+     * Add a credit card to the referral's account
+     * @param {string} referralApiKey - The referral user's production API key.
+     * @param {object} creditCardDetails  - Credit card details:
+     * @param {string} creditCardDetails.number  - Credit card number
+     * @param {string} creditCardDetails.expirationMonth  - Credit card expiration month
+     * @param {string} creditCardDetails.expirationYear  - Credit card expiration year
+     * @param {string} creditCardDetails.cvc  - Credit card CVC
+     * @param {string} primaryOrSecondary  - Whether to add the card as the primary or secondary card (defaults to primary)
+     * @returns {Promise<never>}
+     */
+    static async addCreditCard(
+      referralApiKey,
+      creditCardDetails = {},
+      primaryOrSecondary = 'primary',
+    ) {
+      const stripeCreditCardId = await Referral.#sendCardDetailsToStripe(creditCardDetails);
+      const paymentMethod = await Referral.#sendCardDetailsToEasyPost(
+        referralApiKey,
+        stripeCreditCardId,
+        primaryOrSecondary,
+      );
+      return paymentMethod;
+    }
+
+    static async #getStripeKey() {
+      const response = await api.get('partners/stripe_public_key');
+      return response.body.public_key;
+    }
+
+    /*
+     * Send the credit card details to Stripe to get a Stripe credit card ID.
+     * @param {object} creditCardDetails  - Credit card details:
+     * @param {string} creditCardDetails.number  - Credit card number
+     * @param {string} creditCardDetails.expirationMonth  - Credit card expiration month
+     * @param {string} creditCardDetails.expirationYear  - Credit card expiration year
+     * @param {string} creditCardDetails.cvc  - Credit card CVC
+     * @returns {Promise<string>} - Stripe credit card ID
+     */
+    static async #sendCardDetailsToStripe(creditCardDetails = {}) {
+      const { number, expirationMonth, expirationYear, cvc } = creditCardDetails;
+      const stripeKey = await Referral.#getStripeKey();
+
+      const req = superagent.post(this.#stripeCreditCardUrl).set({
+        Authorization: `Bearer ${stripeKey}`,
+        'Content-Type': 'application/x-www-form-urlencoded',
```

Comment:
Can we include a comment here as to why we need to form encode it. Someone may try to swap it to JSON in the future without us calling it out here.